File: sqry-lang-kotlin/tests/fixtures/relation_tracking/sample.kt
package demo.app

import demo.app.Extensions.trimAll
import demo.lib.Repository as RepoLib
import demo.util.*

class Deferred<T>(val value: T)

object RepoLib {
    suspend fun fetch(value: Int): Deferred<Int> = Deferred(value)
}

object Extensions {
    fun String.trimAll(): String = this.trim()
}

class Service {
    suspend fun process(value: Int): Deferred<Int> {
        val result = RepoLib.fetch(value)
        return result
    }

    fun format(name: String): String {
        return name.trimAll()
    }
}

File: sqry-lang-csharp/tests/fixtures/relation_tracking/sample.cs
using System.Threading.Tasks;
using static System.Math;
using Json = System.Text.Json.JsonSerializer;

namespace Demo.App {
    public static class Extensions {
        public static string TrimAll(this string value) => value.Trim();
    }

    public sealed class User {
        public string Name { get; }
        public User(string name = "anon") {
            Name = name;
        }
    }

    public static class Repository {
        public static Task<User> LoadAsync() => Task.FromResult(new User("repo"));
    }

    public class Service {
        public async Task<User> FetchAsync() {
            var user = await Repository.LoadAsync();
            return user;
        }

        public string Format(string input) {
            return input.TrimAll();
        }
    }
}
